crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/ContributionService.java

```diff
@@ -47,4 +49,45 @@ public AssessmentResponseDTO getAssessmentResult(AssessmentRequestDTO request) {
         }
         return response;
     }
+
+    public boolean checkReassessment(final int repId, final String laaTransactionId) {
+        log.info("Check if reassessment is required for REP_ID={}", repId);
+
+        RepOrderDTO repOrderDTO = maatCourtDataService.getRepOrderByRepId(repId, laaTransactionId);
+        long contributionCount = maatCourtDataService.getContributionCount(repId, laaTransactionId);
+        List<FinancialAssessmentDTO> financialAssessments = repOrderDTO.getFinancialAssessments();
+        List<PassportAssessmentDTO> passportAssessments = repOrderDTO.getPassportAssessments();
+
+        boolean finAssReassFlag = checkFinancialReassessment(contributionCount, financialAssessments);
```

Comment:
As both these methods are checking for contributionCount > 0 condition, we can return false if the contributionCount is 0 and only execute rest of the logic if there any contributions.